### DISCUSSION TITLE
When developing, how to integrate `catch2` with clion?

### DISCUSSION THREAD

**Discussion by StLeoX** [duckdb/duckdb#8467] [Q&A]
Especially, which `Target` should be here?
![2023-08-04_17-27-48](https://github.com/duckdb/duckdb/assets/59193798/4cdaf8e6-3ebf-4ca5-8d44-6af4b35995c8)

**Comment by EpsilonPrime** [comment]
I usually test with unittest there.

**Comment by thomaskejser** [comment]
Once you build the stack, you should see the target in the dropdown. I assume you are building a standalone executable for the test?

I have found that you sometimes need to reload Clion for it to pick up newly added `catch2` test files (reloading the Cmake project might work too). You also need to make sure you call `catch_discover_tests`

Here is a sample `CMakeLists.txt` that works for me in Clion and gives me the `test_generator` option in the dropdown

```
project(test_generator LANGUAGES CXX)

enable_testing()
find_package(Catch2 CONFIG REQUIRED)
add_executable(test_generator TpcH.cpp)
target_link_libraries(test_generator PRIVATE Catch2::Catch2WithMain generator generator_interface)
add_test(NAME test_generator COMMAND test_generator )
include(CTest)
include(Catch)
catch_discover_tests(test_generator)
```
